torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -420,7 +437,7 @@ auto FunctionParameter::check(PyObject* obj, std::vector<py::handle> &overloaded
       // if a size is specified (e.g. IntArrayRef[2]) we also allow passing a single int
       return size > 0 && THPUtils_checkLong(obj);
     }
-    case ParameterType::FLOAT_LIST: return (PyTuple_Check(obj) || PyList_Check(obj));
+    case ParameterType::FLOAT_LIST: return is_float_or_complex_list(obj);
```

Comment:
So I haven't found where the list consisting of float and complex numbers would actually error out, the function signature is `at::ArrayRef<double>`, can you check what happens in this case?

Reply:
In case of complex input, we will get an error. here is a [test](https://github.com/pytorch/pytorch/pull/45673/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.py#diff-b300fed36e386acda9146c82f11d6090R391-R413)

Reply:
Awesome, thanks for checking!